multiqc/modules/samtools/samtools.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python
+
+""" MultiQC module to parse output from samtools stats command"""
+
+from __future__ import print_function
+from collections import OrderedDict
+import logging
+import re
+
+from multiqc import config, BaseMultiqcModule
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+class MultiqcModule(BaseMultiqcModule):
+
+    def __init__(self):
+
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(name='samtools',
+        anchor='samtools', target='samtools',
+        href='http://www.htslib.org',
+        info="Samtools is a suite of programs for interacting"\
+        " with high-throughput sequencing data.")
+        self.samtools_data = dict()
+        for f in self.find_log_files(config.sp['samtools']):
+            s_name = f['s_name'].split(".txt")[0]
+            parsed_data = self.parse_samtools_report(f['f'])
+            if parsed_data is not None:
+                if s_name in self.samtools_data:
+                    log.debug("Duplicate sample name found! Overwriting: {}".format(f['s_name']))
+                self.add_data_source(f)
+                self.samtools_data[s_name] = parsed_data
+
+        if len(self.samtools_data) == 0:
+            log.debug("Could not find any reports in {}".format(config.analysis_dir))
+            raise UserWarning
+
+        log.info("Found {} reports".format(len(self.samtools_data)))
+
+        # Write parsed report data to a file
+        self.write_data_file(self.samtools_data, 'multiqc_samtools')
+
+        # Basic Stats Table
+        # Report table is immutable, so just updating it works
+        self.samtools_stats_table()
+
+
+    def parse_samtools_report (self, raw_data):
+        """ Parse the samtools log file. """
+
+        regexes = {
+                'raw_total_seqs': r"raw\stotal\ssequences:\s+(\d+)",
+            'reads_mapped': r"reads\smapped:\s+(\d+)",
+            'non_primary': r"non-primary\salignments:\s+(\d+)",
+            'reads_paired': r"reads\smapped\sand\spaired:\s+(\d+)",
+            'reads_duplicated': r"reads\sduplicated:\s+(\d+)",
+            'error_rate': r"error\srate:\s+(\d+\.\d+e-\d+)\s+#",
+            'pairs_on_diff_chrom': r"pairs\son\sdifferent\schromosomes:\s+(\d+)"
+        }
```

Comment:
The test data file you uploaded looks really nice - loads of good numbers in there! This is totally up to you, but it could be good to parse _all_ of the stats in the `# Summary Numbers` section here, even if they're not used in the MultiQC report. They'll go into the MultiQC parsed data files and could help people with downstream analysis...